client/client.go

```diff
@@ -791,11 +791,11 @@ func signTargets(updates map[string][]byte, repo *tuf.Repo, initialPublish bool)
 	// iterate through all the targets files - if they are dirty, sign and update
 	for roleName, roleObj := range repo.Targets {
 		if roleObj.Dirty || (roleName == data.CanonicalTargetsRole && initialPublish) {
-			targetsJSON, err := serializeCanonicalRole(repo, roleName, nil)
+			targetsJSON, err := serializeCanonicalRole(repo, data.RoleName(roleName), nil)
```

Comment:
I think `repo.Targets` is already a map of `data.RoleName` to targets, so this cast may not be necessary.